freqtrade/freqai/data_kitchen.py

```diff
@@ -460,6 +460,18 @@ def slice_dataframe(self, timerange: TimeRange, df: DataFrame) -> DataFrame:
 
         return df
 
+    def check_pred_labels(self, df_predictions: DataFrame) -> None:
+        """
+        Check that prediction feature labels match training feature labels.
+        :params:
+        :df_predictions: incoming predictions
+        """
+        train_labels = self.data_dictionary["train_features"].columns
+        pred_labels = df_predictions.columns
+        if len(train_labels.difference(pred_labels)) != 0:
+            self.data_dictionary["prediction_features"] = df_predictions[train_labels]
```

Comment:
Can you put a log warning here, so that we as devs can see when this starts cropping up in bug reports (also doesn't hurt for the user to realize something "went wrong")

```
logger.warning(f'Removed {num_diffs} features from prediction features, these were likely considered constant values during most recent training')
```

Then you can assert the log existence in the tests that you edited (e.g. inside `test_start_backtesting()`, note: ensure to add `caplog` to the arguments of `test_start_backtesting(caplog)`) :

```
    assert log_has_re(
        "Removed 5 features from prediction features",
        caplog,
    )
 ```

Reply:
Adding the log warning and assert in the test showed that the shape check was placed incorrectly: the shape check was made after normalization which caused key errors. I now moved the check to `filter_features()` in `data_kitchen` which means it gets done before normalization.

Also added an assert log in `test_start_backtesting()` about dropping the constant features from the test data.